src/main/java/formflow/library/ScreenController.java

```diff
@@ -146,6 +155,22 @@ ModelAndView getScreen(
     return new ModelAndView("%s/%s".formatted(flow, screen), model);
   }
 
+  /**
+   * Checks if current screen condition is met.
+   *
+   * @param uuid          The uuid of a subflow entry
+   * @param currentScreen The current screen to check
+   * @param submission    submission
+   * @return True - current screen does not meet the condition; False - otherwise
+   */
+  private boolean shouldRedirectToNextScreen(String uuid, ScreenNavigationConfiguration currentScreen, Submission submission) {
+    if (conditionManager.conditionExists(currentScreen.getCondition())) {
+      boolean skipSubflowScreen = currentScreen.getSubflow() != null && !conditionManager.runCondition(currentScreen.getCondition(), submission, uuid);
+      return skipSubflowScreen || !conditionManager.runCondition(currentScreen.getCondition(), submission);
+    }
```

Comment:
I think I'm a little confused by this code here.   Maybe the check for if we are in a subflow could be done first here?  

For example, if we are in a subflow and the condition returns false, then it doesn't make sense to try to run the condition again as if it's not in a subflow.

Reply:
```
 private boolean shouldRedirectToNextScreen(String uuid, ScreenNavigationConfiguration currentScreen, Submission submission) {
    if (conditionManager.conditionExists(currentScreen.getCondition())) {
      if (currentScreen.getSubflow() != null) { 
         return !conditionManager.runCondition(currentScreen.getCondition(), submission, uuid)
      }
      return !conditionManager.runCondition(currentScreen.getCondition(), submission);
    }
```

(Outside of this ticket, this makes me realize we could use a currentScreen.isInSublfow() or something like that)

Reply:
Yeah, that second comment is a lot easier to read